cmd/clusterctl/cmd/migrate.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/core/v1beta2"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/migrate"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/scheme"
+)
+
+type migrateOptions struct {
+	output    string
+	toVersion string
+}
+
+var migrateOpts = &migrateOptions{}
+
+var migrateCmd = &cobra.Command{
+	Use:   "migrate [SOURCE]",
+	Short: "EXPERIMENTAL: Migrate cluster.x-k8s.io resources between API versions",
+	Long: `EXPERIMENTAL: Migrate cluster.x-k8s.io resources between API versions.
+
+This command is EXPERIMENTAL and currently supports only cluster.x-k8s.io core resources
+(Cluster, MachineDeployment, etc.) with v1beta2 as the only supported target version.
+
+SCOPE AND LIMITATIONS (Experimental Status):
+- Only cluster.x-k8s.io resources are converted
+- Other CAPI API groups are passed through unchanged
+- Currently only supports v1beta2 as target version (--to-version flag must be "v1beta2")
+- ClusterClass patches are not migrated
+- Field order may change and comments will be removed in output
+- API version references are dropped during conversion (except cluster class and external
+  remediation references)
+
+Examples:
+  # Migrate from file to stdout
+  clusterctl migrate cluster.yaml
+
+  # Migrate from stdin to stdout
+  cat cluster.yaml | clusterctl migrate
+
+  # Explicitly specify target version (currently only v1beta2 is supported)
+  clusterctl migrate cluster.yaml --to-version v1beta2 --output migrated-cluster.yaml`,
+
+	Args: cobra.MaximumNArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return runMigrate(args)
+	},
+}
+
+func init() {
+	migrateCmd.Flags().StringVarP(&migrateOpts.output, "output", "o", "", "Output file path (default: stdout)")
+	migrateCmd.Flags().StringVar(&migrateOpts.toVersion, "to-version", "v1beta2", "Target API version for migration (currently only v1beta2 is supported)")
+
+	RootCmd.AddCommand(migrateCmd)
+}
+
+func runMigrate(args []string) error {
+	if migrateOpts.toVersion != clusterv1.GroupVersion.Version {
+		return fmt.Errorf("invalid --to-version value %q: currently only %s is supported", migrateOpts.toVersion, clusterv1.GroupVersion.Version)
```

Comment:
please use everywhere github.com/pkg/errors.Errof, Wrap, WithMessage,  instead of fmt.Errorf

Reply:
fixed!